src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -106,7 +106,7 @@ protected function doHeloCommand(): void
 
         /** @var SocketStream $stream */
         $stream = $this->getStream();
-        if (!$stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
+        if ($stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
```

Comment:
I think this is wrong. STARTTLS is not needed when the connection is already using TLS.